packages/discord.js/src/structures/GuildOnboardingPromptOption.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+const { Collection } = require('@discordjs/collection');
+const Base = require('./Base');
+
+/**
+ * Represents the data of an option from a prompt of a guilds onboarding.
+ * @extends {Base}
+ */
+class GuildOnboardingPromptOption extends Base {
+  constructor(client, data, guildId) {
+    super(client);
+
+    const guild = this.guild;
+
+    /**
+     * The id of the guild this onboarding prompt option is from
+     * @type {Snowflake}
+     */
+    this.guildId = guildId;
+
+    /**
+     * The id of the option
+     * @type {Snowflake}
+     */
+    this.id = data.id;
+
+    /**
+     * The channels opted in when this option is selected
+     * @type {Collection<Snowflake, GuildChannel>}
+     */
+    this.channels = data.channel_ids.reduce((channels, channelId) => {
+      channels.set(channelId, guild.channels.cache.get(channelId));
+      return channels;
+    }, new Collection());
+
+    /**
+     * The roles assigned when this option is selected
+     * @type {Collection<Snowflake, Role>}
+     */
+    this.roles = data.role_ids.reduce((roles, roleId) => {
+      roles.set(roleId, guild.roles.cache.get(roleId));
+      return roles;
+    }, new Collection());
+
+    /**
+     * The data for an emoji of a guilds onboarding prompt option
+     * @typedef {Object} GuildOnboardingPromptOptionEmoji
+     * @property {?Snowflake} id The id of the emoji
+     * @property {?string} name The name of the emoji
+     */
+
+    /**
+     * The emoji of the option
+     * @type {GuildOnboardingPromptOptionEmoji}
+     */
+    this.emoji = {
+      id: data.emoji_id,
+      name: data.emoji_name,
+    };
```

Comment:
If both  properties are null ( so it doesn't have an emoji )
shouldn't whole emoji be null? *from a practical point of view*

Reply:
i agree

Reply:
i agree